src/svcli/svcli.cpp

```diff
@@ -122,4 +162,29 @@ int main(int argc, char const *argv[]) {
   }
 
   peparse::DestructParsedPE(pe);
+  return 0;
+}
+
+int main(int argc, char const *argv[]) {
+  argh::parser cmdl(argv);
+  bool extract = false;
+
+  if (cmdl[{"-v", "--version"}]) {
+    std::cout << "svcli (uthenticode) version " << UTHENTICODE_VERSION << '\n';
+    return 0;
+  } else if (cmdl[{"-x", "--extract"}]) {
+    extract = true;
+  } else if (cmdl[{"-h", "--help"}] || argc != 2) {
+    std::cout << "Usage: svcli [options] <file>\n\n"
+              << "Options:\n"
+              << "\t-v, --version\tPrint the version and exit\n"
+              << "\t-x, --extract\tExtract the first certificate blob\n"
+              << "\t-h, --help\tPrint this help message and exit\n\n"
+              << "Arguments:\n"
+              << "\t<input-file>\tThe PE to parse for Authenticode data\n"
+              << "\t[output-file]\tWith -x/--extract the file to dump the buffer into (leave empty "
+                 "or use - for stdout)\n";
+    return 0;
+  }
+  return realmain(cmdl, extract);
```

Comment:
Rather than breaking this into `realmain`, can we keep this all as one function? The `realmain` name isn't super descriptive + passing in both a `argh::parser` *and* a partially extracted result (`extract`) makes it look like a leaky abstraction.

Reply:
I'm not quite clear what it is you want. Disposing of `realmain` I got. But the remainder would naturally disappear then, no?

Reply:
Yeah -- sorry for the confusion, the second part was just me thinking out loud about why I'd prefer just one function here 🙂